addon/webextension/selector/ui.js

```diff
@@ -91,6 +91,31 @@ this.ui = (function() { // eslint-disable-line no-unused-vars
     return result;
   }
 
+  function isDownloadOnly() {
+    return window.incognito;
+  }
```

Comment:
I think we want the background page to control/detect this, not the content script. Or at least we have a couple other instances like some privacy settings where we might disable it. (If it's hard this could be a followup bug)

Reply:
Yeah, per my last comment, the background page does detect and assign this state. Borrowing 'incognito' from Tabs.tab probably made this harder to understand at a glance.

I think this should be ok from a security standpoint, as the incognito state isn't leaked to scripts running in the underlying page:

<img width="919" alt="screen shot 2017-10-23 at 11 21 28 am" src="https://user-images.githubusercontent.com/96396/31905875-5e28c48e-b7e4-11e7-861d-e542fa5a1b60.png">

Reply:
Oh, I get it then. Maybe a name like downloadOnly would be better? Also you can assign it to `this` which for subtle reasons I don't understand is slightly cleaner than `window`.

Reply:
Updated. Oddly, assigning to `this` instead of `window` doesn't seem to work, so I've left that with an explanatory note.

Reply:
Oh sure, that whole thing was a little superstitious anyway (well, based on feedback from @kmaglione, I think based around the levels of wrappers and performance, which wouldn't really matter here). Also of course in the module closures `this` gets mapped to something else, so it's probably all just a bit of a nuisance.